### PR TITLE
Storybook: Upgrade to the latest version (v8.4.7)

### PR DESCRIPTION
**Pull request by tyxla**
## What?
Upgrading Storybook to the latest version (v8.4.7). 

## Why?
We've been stuck on Storybook 7.x for a while. In #67574 we upgraded to v8.0, and this PR brings Storybook to the latest version.

## How?
It's mostly bumping the versions, which results in a major dependency cleanup (see #67840 that attempted to do parts of that). 

The only non-dependency change is the deprecation of `autodocs`, see https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mainjs-docsautodocs-is-deprecated

## Testing Instructions
* Verify storybook builds
* Test every component or docs page to ensure nothing breaks.

### Testing Instructions for Keyboard
Same

## Screenshots or screencast 

None

### CODE REVIEW

File: storybook/preview.js [hunk 1]
```diff
@@ -155,3 +155,5 @@ export const parameters = {
 	},
 	sourceLinkPrefix: 'https://github.com/WordPress/gutenberg/blob/trunk/',
 };
+
+export const tags = [ 'autodocs' ];
```
Comment (ciampo) [inline storybook/preview.js:159]: The [migrations docs](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mainjs-docsautodocs-is-deprecated) suggesting adding a default export, is there a reason why we're not following that suggestion and opting for a named export instead? Could this ever stop working in the future?

**Review by ciampo — APPROVED** [review]
Some tested a few stories, so far everything is looking good!

Screenshot 2024-12-12 at 10 57 10

I love this filtering feature! We should definitely consider using it more.

As a follow-up, I wish we could get rid of all the remaining console warnings:

Screenshot 2024-12-12 at 11 04 07

File: storybook/preview.js (hunk 1 above)
Comment (tyxla) [reply storybook/preview.js:159]: The auto migrator didn't have a problem catching that and working with it. I think it shouldn't be an issue. If it does, we'll find out quickly enough.

**Comment by tyxla** [comment]
Thanks for taking a look and testing 🙇‍♂️ 

> I love this filtering feature! We should definitely consider using it more.

Yes! Might motivate us to further adopt tags in our stories.

> As a follow-up, I wish we could get rid of all the remaining console warnings:

Yup, those are pre-existing, and I'd love to clean them all soon. One step at a time!

**Pull request merged**